frontend/src/components/editor-page/editor-pane/hooks/yjs/use-send-realtime-activity.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { useIsDocumentVisible } from '../../../../../hooks/common/use-is-document-visible'
+import type { MessageTransporter } from '@hedgedoc/commons'
+import { MessageType } from '@hedgedoc/commons'
+import { useEffect } from 'react'
+
+/**
+ * Sends the activity state (based on the fact if the tab is focused) to the backend.
+ *
+ * @param messageTransporter The message transporter that handles the connection to the backend
+ */
+export const useSendRealtimeActivity = (messageTransporter: MessageTransporter) => {
+  const active = useIsDocumentVisible()
```

Comment:
since we already use react-use hooks in our project, we could do a combination of `useIsDocumentVisible` and [`useIsIdle`](https://github.com/streamich/react-use/blob/master/docs/useIdle.md) with a idle-time of for example 300 seconds.